### PR TITLE
[https://nvbugs/5934461][fix] Propagate logits from prefill to decode in disagg

### PR DESCRIPTION
**Pull request by brb-nv**
## Description

This is to address the following bug by propagating logits from prefill to decode in disagg mode.
https://nvbugspro.nvidia.com/bug/5934461

## Test Coverage

```
$ pytest tests/integration/defs/disaggregated/test_disaggregated_single_gpu.py::test_disaggregated_logits[True-TinyLlama-1.1B-Chat-v1.0] -s -v
$ pytest tests/unittest/disaggregated/test_openai_disagg_service.py::TestFirstGenLogitsSerializeRoundtrip -s -v
$ pytest tests/unittest/_torch/executor/test_chunked_logits.py::TestGetLatestLogitsUnexcluded -s -v
```

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

## Release Notes

* **New Features**
  * Disaggregated generation logits are now available and properly propagated through prefill and generation phases
  * Added streaming support for disaggregated generation

* **Bug Fixes**
  * Fixed generation logits handling in context-only disaggregated responses

* **Tests**
  * Added integration tests validating disaggregated logits for streaming and non-streaming scenarios

### CODE REVIEW

File: disagg_logits_repro.py
Comment (pcastonguay) [inline disagg_logits_repro.py]: Opening thread to not forget to remove.

**Review by pcastonguay — APPROVED** [review]

**Review by chzblych — APPROVED** [review]

**Comment by pcastonguay** [comment]
Looks like some user buffer tests are failing. Could you take a closer look? Thanks.

**Review by StanleySun639 — APPROVED** [review]

File: tests/integration/test_lists/test-db/l0_h100.yml [hunk 1]
```diff
@@ -133,6 +133,10 @@ l0_h100:
   - disaggregated/test_disaggregated_single_gpu.py::test_disaggregated_llama_context_capacity[False-False-DeepSeek-V3-Lite-fp8/fp8]
   - disaggregated/test_disaggregated_single_gpu.py::test_disaggregated_spec_dec_batch_slot_limit[True-False-EAGLE3-LLaMA3.1-Instruct-8B-Llama-3.1-8B-Instruct]
   - disaggregated/test_disaggregated_single_gpu.py::test_disaggregated_spec_dec_batch_slot_limit[False-False-EAGLE3-LLaMA3.1-Instruct-8B-Llama-3.1-8B-Instruct]
+  - disaggregated/test_disaggregated_single_gpu.py::test_disaggregated_logits[False-TinyLlama-1.1B-Chat-v1.0]
```
[the remark this answers is not in the document]
Comment (brb-nv) [reply tests/integration/test_lists/test-db/l0_h100.yml]: Done. Removed from `l0_a10.yaml`.
https://github.com/NVIDIA/TensorRT-LLM/pull/11767/commits/10eff5f426cd0ac6d1859e2c1e8d459041bc3807
